internal/subshell/cmd/env.go

```diff
@@ -69,9 +69,14 @@ func (c *CmdEnv) unset(keyName, oldValue string) error {
 }
 
 func cleanPath(keyValue, oldEntry string) string {
+	oldEntries := make(map[string]bool)
+	for _, entry := range strings.Split(oldEntry, string(os.PathListSeparator)) {
+		oldEntries[filepath.Clean(entry)] = true
+	}
+
 	var newValue []string
 	for _, entry := range strings.Split(keyValue, string(os.PathListSeparator)) {
-		if filepath.Clean(entry) == filepath.Clean(oldEntry) {
+		if _, ok := oldEntries[filepath.Clean(entry)]; ok {
```

Comment:
Nit:
```suggestion
		if oldEntries[filepath.Clean(entry)] {
```

Since you're using `map[string]bool` you can do this. If you want to keep your old pattern then I _think_ `map[string]struct{}` is more performant.

Reply:
Thanks for the tip. I did not know `map[string]bool` was special.